### ISSUE TITLE
Throw proper error on order when data source is unavailable 

### ISSUE THREAD

**Issue by KatunaNorbert** [oceanprotocol/ocean.js#850]
**Describe the bug**
When calling the `ocean.asset.order()` and the data source is unavailable the following message is returned:
![request_error](https://user-images.githubusercontent.com/37236152/122368656-32c02880-cf66-11eb-95c2-40a156944fc0.png)

**To Reproduce**
It can be reproduced by ordering the asset with the following did: did:op:08A814050DBC4DA746B3b608FaC1C72d13d01f4C
This asset is found on Rinkeby network

**Expected behavior**
The following error should be returned when data source is not available:
![Screenshot 2021-06-17 at 12 22 31](https://user-images.githubusercontent.com/37236152/122369408-d6113d80-cf66-11eb-8904-121ba06f58c9.png)

**Comment by jamiehewitt15** [comment]
@KatunaNorbert Did you purchase the asset from the market? it's currently unavailable for purchase 

![Datatoken unavailable 2021-06-30_14-46](https://user-images.githubusercontent.com/20739535/123955266-17b3d680-d9b2-11eb-9e96-f62e0556a4eb.png)

**Comment by KatunaNorbert** [comment]
You can't buy it because you will get the error mentioned above. But the button should be enabled:
![Screenshot 2021-06-30 at 14 53 07](https://user-images.githubusercontent.com/37236152/123956185-12a35700-d9b3-11eb-90f8-ba0d98a408c4.png)

**Comment by jamiehewitt15** [comment]
hmm that's odd... I'm not sure why the buy button isn't enabled when I view it

**Comment by jamiehewitt15** [comment]
Currently I'm getting this error when trying to purchase the dataset, which is different to the one described in the issue: 
![Datatoken purchase unavailable 2021-06-30_17-34](https://user-images.githubusercontent.com/20739535/123980157-fd85f280-d9c9-11eb-93f3-47c7a456689c.png)
 

Currently `isConsumable` is set to `false` so I had to manually enable the purchase button on a local version of the market to try and purchase the asset.

**Comment by KatunaNorbert** [comment]
This is for buying dataTokens?

**Comment by mihaisc** [comment]
Not sure there are any more datatokens in that pool, so you can't really buy

**Comment by bogdanfazakas** [comment]
yes the error above appears when there are not enough datatokens in the pool available to buy, users should not bump into it when we merge https://github.com/oceanprotocol/market/pull/627 into main

**Comment by jamiehewitt15** [comment]
@KatunaNorbert @bogdanfazakas Is there anyway to reproduce the error that was mentioned in the issue `HTTP request failed`? Or is there another asset you know of that's got the same issue?

**Comment by jamiehewitt15** [comment]
Ok, I managed to recreate this error now. This is what I'm getting in the console: 

![error 2021-07-05_14-12](https://user-images.githubusercontent.com/20739535/124463200-355eb280-dd9b-11eb-94ec-4f4ba4bcd4bd.png)
